File: app/src/main/java/knf/kuma/videoservers/FireServer.java
package knf.kuma.videoservers;

import android.content.Context;
import android.support.annotation.Nullable;

import org.jsoup.Jsoup;

import knf.kuma.commons.PatternUtil;

import static knf.kuma.videoservers.VideoServer.Names.FIRE;

public class FireServer extends Server {
    public FireServer(Context context, String baseLink) {
        super(context, baseLink);
    }

    @Override
    public boolean isValid() {
        return baseLink.contains("efire.php");
    }

    @Override
    public String getName() {
        return FIRE;
    }

    @Nullable
    @Override
    VideoServer getVideoServer() {
        try {
            String frame = PatternUtil.extractLink(baseLink);
            String media_func = Jsoup.connect(frame).get().select("script").last().outerHtml();
            String download = Jsoup.connect(PatternUtil.extractMediaLink(media_func)).get().select("a[href~=http://download.*]").first().attr("href");
            return new VideoServer(FIRE, new Option(getName(), null, download));
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/knf/kuma/videoservers/FireServer.kt
package knf.kuma.videoservers

import android.content.Context
import knf.kuma.commons.PatternUtil
import knf.kuma.commons.jsoupCookies
import knf.kuma.videoservers.VideoServer.Names.FIRE
import org.jsoup.Jsoup

class FireServer internal constructor(context: Context, baseLink: String) : Server(context, baseLink) {

    override val isValid: Boolean
        get() = baseLink.contains("efire.php")

    override val name: String
        get() = FIRE

    override val videoServer: VideoServer?
        get() {
            return try {
                val frame = PatternUtil.extractLink(baseLink)
                val mediaFunc = jsoupCookies(frame).get().select("script").last().outerHtml()
                val download = Jsoup.connect(PatternUtil.extractMediaLink(mediaFunc)).get().select("a[href~=http://download.*]").first().attr("href")
                VideoServer(FIRE, Option(name, null, download))
            } catch (e: Exception) {
                e.printStackTrace()
                null
            }

        }
}